pytorch_lightning/trainer/optimizers.py

```diff
@@ -205,6 +220,13 @@ def __repr__(self):
         return "No Optimizer"
 
 
+def _validate_optim_conf(optim_conf: Dict[str, Any]) -> None:
+    valid_keys = {"optimizer", "lr_scheduler", "frequency", "monitor"}
```

Comment:
Something doesn't look right here. Based on https://pytorch-lightning.readthedocs.io/en/latest/api/pytorch_lightning.core.lightning.html#pytorch_lightning.core.lightning.LightningModule.configure_optimizers we're expecting something like this:

```
"optimizer": optimizer,
        "lr_scheduler": {
            "scheduler": ReduceLROnPlateau(optimizer, ...),
            "monitor": "metric_to_track",
            "frequency": 5,
        }
```

This function seems to assume all keys will show up on the same level.

Reply:
yep, this function just checks the optimizer level keys.. doesn't go inside the scheduler config

Reply:
Are frequency and monitor even legal on this level? I only ever saw them used inside the scheduler config.

And can we also validate the scheduler config?

Reply:
yes monitor is legal at this level, although not a big fan.. makes it a bit confusing IMO for some..
frequency here belongs to the optimizer.. usually used in case of multiple optimizers.
yes we have validation of scheduler config as well here: https://github.com/PyTorchLightning/pytorch-lightning/blob/3392215ef69b483bff82bf7b0270c2349b33a73c/pytorch_lightning/trainer/optimizers.py#L150-L152

Reply:
Okay, that works. Thanks for the clarification!